business/workloads.go

```diff
@@ -1718,11 +1718,15 @@ func fetchWorkload(ctx context.Context, layer *Layer, criteria WorkloadCriteria)
 		// Add the Proxy Status to the workload
 		for _, pod := range w.Pods {
 			if pod.HasIstioSidecar() {
-				ps, err := layer.ProxyStatus.GetPodProxyStatus(criteria.Namespace, pod.Name)
-				if err != nil {
-					log.Warningf("GetPodProxyStatus is failing for [namespace: %s] [pod: %s]: %s ", criteria.Namespace, pod.Name, err.Error())
+				if config.Get().ExternalServices.Istio.IstioApiEnabled {
+					ps, err := layer.ProxyStatus.GetPodProxyStatus(criteria.Namespace, pod.Name)
+					if err != nil {
+						log.Warningf("GetPodProxyStatus is failing for [namespace: %s] [pod: %s]: %s ", criteria.Namespace, pod.Name, err.Error())
+					}
+					pod.ProxyStatus = castProxyStatus(ps)
+				} else {
+					pod.ProxyStatus = &models.ProxyStatus{}
```

Comment:
This might actually port forward or exec directly into the pod to get the proxy status instead of talking with the istiod. Have you tried this without istiod access and it doesn't work?

Reply:
Thanks, @nrfox, this is a good suggestion. I would try to get the information like this.

Reply:
Although this may work, I'm a little worried about perf when scaling to a lot of pods.  Is that something we should worry about?  Do we already exec into the pod for some other reason?  For me, if the users elect to block istiod access perhaps they can expect to lose this  feature.

Reply:
> For me, if the users elect to block istiod access perhaps they can expect to lose this feature.

@jshaughn this is what I'm wondering. I wasn't sure if the proxy status came from the pod's proxy or from the istiod pod but it looks like it comes from the istiod pod: https://github.com/kiali/kiali/blob/2165abf70c4cfae7d4a08665af6a008329a35ab3/kubernetes/istio.go#L254-L275. In which case this feature _should_ be disabled when istiod access is disabled. Sorry for the noise!

Reply:
no worries, it was a good question.